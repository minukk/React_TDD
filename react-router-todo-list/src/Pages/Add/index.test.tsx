import React from 'react';
import { Router, useLocation } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { screen, render, fireEvent } from '@testing-library/react';
import 'jest-styled-components';

import { Add } from './index';
import { ToDoListProvider } from 'Context';

describe('<App />', () => {
  it('renders component correctly', () => {
    const history = createMemoryHistory();
    history.push('/add');

    const { container } = render(
      <Router navigator={history} location={history.location}>
        <Add />
      </Router>,
    );

    const input = screen.getByPlaceholderText('할 일을 입력해 주세요.');
    expect(input).toBeInTheDocument();
    const button = screen.getByText('추가');
    expect(button).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('add a new ToDo and redirect to the root page', () => {
    const history = createMemoryHistory();
    history.push('/add');

    const TestComponent = () => {
      const { pathname } = useLocation();
      return (
        <ToDoListProvider>
          <div>{pathname}</div>
          <Add />
        </ToDoListProvider>
      );
    };

    render(
      <Router navigator={history} location={history.location}>
        <TestComponent />
      </Router>,
    );

    const pathName = screen.getByText('/add');
    expect(pathName).toBeInTheDocument();

    const input = screen.getByPlaceholderText('할 일을 입력해 주세요.');
    const button = screen.getByText('추가');

    fireEvent.change(input, { target: { value: 'New ToDo' } });
    fireEvent.click(button);

    expect(history.location.pathname).toBe('/');
    expect(localStorage.getItem('ToDoList')).toBe('["New ToDo"]');
  });
});
