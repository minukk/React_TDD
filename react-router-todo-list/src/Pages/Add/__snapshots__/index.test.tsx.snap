// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders component correctly 1`] = `
.c3 {
  padding: 10px 20px;
  border-radius: 8px;
  background-color: #304ffe;
  text-align: center;
  cursor: pointer;
}

.c3:hover {
  background-color: #1e40ff;
}

.c3:active {
  box-shadow: inset 5px 5px 10px rgba(0,0,0,0.2);
}

.c4 {
  color: #fff;
  font-size: 16px;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 10px;
  border: 1px solid #bdbdbd;
  border-radius: 8px;
  font-size: 16px;
  outline: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px;
  border-radius: 8px;
  background-color: #fff;
  box-shadow: 5px 5px 10px rgba(0,0,0,0.2);
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <input
        class="c2"
        placeholder="할 일을 입력해 주세요."
        value=""
      />
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          추가
        </div>
      </div>
    </div>
  </div>
</div>
`;
